template/androidApp/src/main/java/io/bloco/template/MainActivity.kt

```diff
@@ -18,8 +37,50 @@ class MainActivity : ComponentActivity() {
     @InstallIn(ActivityComponent::class)
     interface ViewModelFactoryProvider : BaseViewModelFactoryProvider
 
+    private val viewModel: MainActivityViewModel by viewModels()
+
     override fun onCreate(savedInstanceState: Bundle?) {
+        val splashScreen = installSplashScreen()
         super.onCreate(savedInstanceState)
-        setContent { TemplateTheme { TemplateNaveHost() } }
+
+        var uiState: MainActivityUiState by mutableStateOf(Loading)
```

Comment:
Why not a plain var?

Reply:
that does not trigger recomposition. does it?

Reply:
The splash screen does not care about compose. It checks the condition by itself I believe.